File: twao/VillageAssignment.java
package twao;

import twao.villages.Village;

public final class VillageAssignment {
    private final Village departure;
    private final Village destination;
    private final int squaredDistance;

    public VillageAssignment(Village departure, Village destination, int squaredDistance) {
        this.departure = departure;
        this.destination = destination;
        this.squaredDistance = squaredDistance;
    }

    @Override
    public String toString() {
        return String.format("%s -> %s", departure, destination);
    }

    /**
     * -----------------------------------------------------------
     * getters and setters
     * -----------------------------------------------------------
     */

    public int getSquaredDistance() { return squaredDistance; }

    public Village getDeparture() { return departure; }

    public Village getDestination() { return destination; }
}

File: src/main/java/io/github/stiv3ns/twactionorganizer/twao/VillageAssignment.kt
package io.github.stiv3ns.twactionorganizer.twao

import io.github.stiv3ns.twactionorganizer.twao.villages.Village
import kotlinx.serialization.Serializable

@Serializable
data class VillageAssignment(
        val departure: Village,
        val destination: Village,
        val squaredDistance: Int
) {
    override fun toString()
        = "$departure -> $destination"
}